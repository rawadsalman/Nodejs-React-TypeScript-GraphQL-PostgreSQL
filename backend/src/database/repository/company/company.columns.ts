export default [
  'company.id',
  'company.uuid',
  'company.name',
  'company.memberSince',
  'company.taxationCountry',
  'company.germanVatNumber',
  'company.subscriptionFee',
  'company.commissionFee',
  'company.marketplaces',
  'company.addressBillingFullName',
  'company.addressBillingLine1',
  'company.addressBillingLine2',
  'company.addressBillingPhone',
  'company.addressBillingCity',
  'company.addressBillingState',
  'company.addressBillingZip',
  'company.addressBillingCountry',
  'company.created_at',
  'company.updated_at',
  'company.registerNumber',
  'company.legalForm',
  'company.personAccountable',
]
