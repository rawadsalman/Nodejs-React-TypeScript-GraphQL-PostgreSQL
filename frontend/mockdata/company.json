{
  "name": "My Good Looking Company",
  "uuid": "00000000-0000-0000-0000-000000000000",
  "memberSince": "2019-11-27",
  "taxationCountry": "GER",
  "germanVatNumber": "1233323",
  "subscriptionFee": 20,
  "commissionFee": 0.5,
  "registerNumber": "ur84u89r",
  "legalForm": "9390ur9",
  "personAccountable": "Bob",
  "accountHolder": "Yasboi Foo",
  "marketplaces": [
    "DE",
    "UK",
    "IT"
  ],
  "addressBillingFullName": "Yasboi Foo",
  "addressBillingLine1": "My Good Looking Company",
  "addressBillingLine2": "Wöhrdstraße 25",
  "addressBillingPhone": "",
  "addressBillingCity": "Tübingen",
  "addressBillingState": "Baden-Württemberg",
  "addressBillingZip": "72070",
  "addressBillingCountry": "Germany",
  "created_at": "2020-05-07 09:41:03",
  "updated_at": "2020-05-07 09:41:03"
}