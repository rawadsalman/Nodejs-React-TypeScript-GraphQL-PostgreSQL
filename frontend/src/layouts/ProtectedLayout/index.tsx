import React, { useEffect } from 'react'
import { useLocation } from 'react-router-dom'
import Sidebar from '../../components/Layout/Sidebar'
import ProtectedRoutes from '../../routes/protected'
import { PageWrapper, SidebarLayoutWrapper, SidebarWrapper } from './styles'

const ProtectedLayout: React.FC = () => {
  let location = useLocation()

  useEffect(() => {
    window.analytics.page()
  }, [location])

  return (
    <SidebarLayoutWrapper>
      <SidebarWrapper width={78}>
        <Sidebar />
      </SidebarWrapper>
      <PageWrapper>
        <ProtectedRoutes />
      </PageWrapper>
    </SidebarLayoutWrapper>
  )
}

export default ProtectedLayout
