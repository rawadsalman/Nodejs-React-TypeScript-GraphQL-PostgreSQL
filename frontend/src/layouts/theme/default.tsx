// rgb(17, 24, 57) blau
// rgb(247, 230, 153)	gelb

const defaultTheme = {
  colors: {
    primary: 'rgb(255, 0, 130)',
    secondary: '#1e272e',
    green: '#16a085',
    darkBlue: '#001529',
    almostWhite: '#f0f0f0',
    almostGrey: '#eaeaea',
    yellow: '#f6e58d',
    paleRed: '#ff7675',
  },
  modal: {
    width: 720,
  },
}

export default defaultTheme
