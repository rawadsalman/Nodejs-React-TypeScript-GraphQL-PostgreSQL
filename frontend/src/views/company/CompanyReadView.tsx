import React from 'react'
import { ScrollWrapper } from '../../components/Layout/ContentWrapper'
import CustomPageHeader from '../../components/Layout/CustomPageHeader'
import { StyledContent } from '../styles'
import { ShowCompanyViewModel } from '../../generated/api'
import { RouteComponentProps, useParams } from 'react-router-dom';
import { GET_COMPANY_DETAILS } from '../../gqlQueries/companies/getOne'
import { useQuery } from '@apollo/client'
import LoadingSpinner from '../../components/Basic/LoadingSpinner'
import {Result, Row ,Descriptions} from 'antd'
import HomeButton from './components/HomeButton'


interface ICompanyReadView extends RouteComponentProps {
   company: ShowCompanyViewModel;
}
const CompanyReadView: React.FC<ICompanyReadView> = ( props ) => {
   
  const uuid=(props as any).match.params.id;
  const { loading, error, data} = useQuery(GET_COMPANY_DETAILS,{ variables: {uuid} })

  const companydetails = data?.company[0]
   if (loading) {
    console.log("loading")
    return <LoadingSpinner data-testid="spinner" />
  }

   if (error || companydetails === null || companydetails === undefined) {
    return (
      <Row justify="center">
        <Result status={'500' as any} title="500" subTitle="Error from the server" />
      </Row>
    )
  } 

  return (
    <ScrollWrapper>
      <CustomPageHeader title={`Company Details`} />
      <StyledContent>
        <Row>
          <Descriptions bordered>

            <Descriptions.Item label="Company Name">{companydetails.name}</Descriptions.Item>
            <Descriptions.Item label="Uuid">{companydetails.uuid}</Descriptions.Item>
            <Descriptions.Item label="Member Since">{companydetails.memberSince}</Descriptions.Item>
            <Descriptions.Item label="Address Billing City">{companydetails.addressBillingCity}</Descriptions.Item>
            <Descriptions.Item label="Address Billing Phone">{companydetails.addressBillingPhone}</Descriptions.Item>

          </Descriptions>
        </Row>
        <br/>
        <br/>
        <Row justify="center">
           <HomeButton />
        </Row>
        
        
      </StyledContent>
    </ScrollWrapper>
  )
}

export default CompanyReadView
