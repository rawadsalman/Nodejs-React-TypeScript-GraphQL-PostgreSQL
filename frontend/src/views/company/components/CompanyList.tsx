import { Button, Col, Input, Row, Table } from 'antd'
import React, { useState } from 'react'
import { Link } from 'react-router-dom'
import { ShowCompanyViewModel } from '../../../generated/api'
import DeleteButton from './DeleteButton'

import AddCompany2 from './AddCompany2'

interface ICompanyList {
  companies: ShowCompanyViewModel[],
}

const columns = [
  {
    title: 'Company Name',
    dataIndex: 'name',
    key: 'companyName',
  },
  {
    title: 'Market Places',
    dataIndex: 'marketplaces',
    key: 'marketplaces',
    render: (places: string[]) => places.join(),
  },
  {
    title: 'Action',
    key: 'action',
    render: (text: string, record: ShowCompanyViewModel) => ( 
      <span>
        <Link to={`/companies/${record.uuid}`} >Open</Link>
        {/* <DeleteButton uuid={`${record.uuid}`}/> */}
  </span>

    ),
  },
  
]

const CompanyList: React.FC<ICompanyList> = ({ companies }) =>
{

 const [showaddform, setshowaddform] = React.useState(false);

return (
  
  <div>
    <Row>
       <Button onClick={(event: React.MouseEvent<HTMLInputElement>) => {
        setshowaddform(true)
       }}>Add Company</Button>
        {showaddform && <AddCompany2 /> }
    </Row>
    <Row>
      <Col xs={24}>
        <Table bordered dataSource={companies} rowKey={'uuid'} columns={columns} pagination={false} />
      </Col>
    </Row>
    
  </div> 
 
);
}
export default CompanyList
