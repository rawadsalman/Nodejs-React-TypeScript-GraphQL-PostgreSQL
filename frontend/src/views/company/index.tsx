/* istanbul ignore file */
import React from 'react'
import { Route, RouteComponentProps, Switch } from 'react-router-dom'
import ListView from './CompanyListView'
import CompanyReadView from './CompanyReadView'

const CompanyRoutes: React.FC<RouteComponentProps<{ id: string }>> = ({ match }) => (
  <Switch>
    <Route exact path={`${match.path}`} component={ListView} />
    <Route path={`${match.path}/:id`} component={CompanyReadView} />
  </Switch>
)

export default CompanyRoutes
