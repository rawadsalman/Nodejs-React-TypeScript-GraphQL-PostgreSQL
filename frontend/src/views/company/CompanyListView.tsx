import { PlusOutlined } from '@ant-design/icons'
import { useQuery } from '@apollo/client'
import { Button, Empty, Result, Row } from 'antd'
import React from 'react'
import { Link, RouteComponentProps } from 'react-router-dom'
import LoadingSpinner from '../../components/Basic/LoadingSpinner'
import { ScrollWrapper } from '../../components/Layout/ContentWrapper'
import CustomPageHeader from '../../components/Layout/CustomPageHeader'
import { ShowCompanyViewModel } from '../../generated/api'
import { GET_ALL_COMPANIES } from '../../gqlQueries/companies/getAll'
import { StyledContent } from '../styles'
import CompanyList from './components/CompanyList'

const CompanyListView: React.FC<RouteComponentProps> = ({ match }) => {
  const { loading, error, data} = useQuery<{ companies: ShowCompanyViewModel[] }>(GET_ALL_COMPANIES)
  const companies = data?.companies
  if (loading) {
    return <LoadingSpinner data-testid="spinner" />
  }

  if (error || companies === null || companies === undefined) {
    return (
      <Row justify="center">
        <Result status={'500' as any} title="500" subTitle="Error from the server" />
      </Row>
    )
  }

  if (companies?.length === 0) {
    return (
      <>
        <Row justify="center">
          <Empty image={Empty.PRESENTED_IMAGE_SIMPLE} />
        </Row>
      </>
    )
  }
  
  return (
    <ScrollWrapper>
      <CustomPageHeader title={'Companies'} />
      <StyledContent>
        <CompanyList companies={companies} />
        {/* <CompanyDetailList companies={companies} /> */}

      </StyledContent>
    </ScrollWrapper>
  )
}

export default CompanyListView
