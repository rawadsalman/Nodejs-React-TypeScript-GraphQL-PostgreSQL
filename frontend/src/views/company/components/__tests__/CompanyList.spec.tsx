import { render } from '@testing-library/react'
import React from 'react'
import CompanyList from '../CompanyList'
import { MemoryRouter } from 'react-router-dom'
import { ThemeProvider } from '@emotion/react'
import { defaultTheme } from '../../../../layouts/theme'
import company from '../../../../../mockdata/company.json'

describe('Company ListView', () => {
  it('matches snapshot', () => {
    const wrapper = render(
      <ThemeProvider theme={defaultTheme}>
        <MemoryRouter>
          {/* <CompanyList companies={[company]} /> */}  
        </MemoryRouter>
      </ThemeProvider>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
