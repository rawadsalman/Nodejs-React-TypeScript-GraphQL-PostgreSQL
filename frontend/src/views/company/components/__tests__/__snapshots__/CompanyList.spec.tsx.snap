// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company ListView matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ant-row"
        >
          <div
            class="ant-col ant-col-xs-24"
          >
            <div
              class="ant-table-wrapper"
            >
              <div
                class="ant-spin-nested-loading"
              >
                <div
                  class="ant-spin-container"
                >
                  <div
                    class="ant-table ant-table-bordered"
                  >
                    <div
                      class="ant-table-container"
                    >
                      <div
                        class="ant-table-content"
                      >
                        <table
                          style="table-layout: auto;"
                        >
                          <colgroup />
                          <thead
                            class="ant-table-thead"
                          >
                            <tr>
                              <th
                                class="ant-table-cell"
                              >
                                Company Name
                              </th>
                              <th
                                class="ant-table-cell"
                              >
                                Market Places
                              </th>
                              <th
                                class="ant-table-cell"
                              >
                                Action
                              </th>
                            </tr>
                          </thead>
                          <tbody
                            class="ant-table-tbody"
                          >
                            <tr
                              class="ant-table-row ant-table-row-level-0"
                              data-row-key="00000000-0000-0000-0000-000000000000"
                            >
                              <td
                                class="ant-table-cell"
                              >
                                My Good Looking Company
                              </td>
                              <td
                                class="ant-table-cell"
                              >
                                DE,UK,IT
                              </td>
                              <td
                                class="ant-table-cell"
                              >
                                <span>
                                  <a
                                    href="/companies/00000000-0000-0000-0000-000000000000"
                                  >
                                    Open
                                  </a>
                                </span>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col ant-col-xs-24"
        >
          <div
            class="ant-table-wrapper"
          >
            <div
              class="ant-spin-nested-loading"
            >
              <div
                class="ant-spin-container"
              >
                <div
                  class="ant-table ant-table-bordered"
                >
                  <div
                    class="ant-table-container"
                  >
                    <div
                      class="ant-table-content"
                    >
                      <table
                        style="table-layout: auto;"
                      >
                        <colgroup />
                        <thead
                          class="ant-table-thead"
                        >
                          <tr>
                            <th
                              class="ant-table-cell"
                            >
                              Company Name
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              Market Places
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              Action
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="ant-table-tbody"
                        >
                          <tr
                            class="ant-table-row ant-table-row-level-0"
                            data-row-key="00000000-0000-0000-0000-000000000000"
                          >
                            <td
                              class="ant-table-cell"
                            >
                              My Good Looking Company
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              DE,UK,IT
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <span>
                                <a
                                  href="/companies/00000000-0000-0000-0000-000000000000"
                                >
                                  Open
                                </a>
                              </span>
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
